main.go

```diff
@@ -259,16 +271,35 @@ func visit(url *url.URL) {
 // readResponseBody consumes the body of the response.
 // readResponseBody returns an informational message about the
 // disposition of the response body's contents.
-func readResponseBody(resp *http.Response) string {
+func readResponseBody(req *http.Request, resp *http.Response) bool {
 	// TODO(dfc) do not process body if status code is in the 30x range
 
 	// TODO(dfc) if we issued a HEAD request, there is no body to process.
 
+	if saveOutput == true || outputFile != "" {
+		var filename string
+
+		if saveOutput == true {
+			parts := strings.Split(req.URL.RequestURI(), "/")
+			filename = parts[len(parts)-1]
+		} else {
+			filename = outputFile
+		}
+
+		fp, err := os.Create(filename)
+		if err != nil {
+			log.Fatalf("unable to create file %s", outputFile)
+		}
+
+		io.Copy(fp, resp.Body)
```

Comment:
There is duplication here. We always want to call `io.Copy`, the only thing that matters is the destination; a temp file, -o from the request, -O from the filename, or io.Discard.